### PR TITLE
Public dev merged

### LINKED ISSUE #71: Genome metadata
### Description of feature

Improve genome metadata fetching and processing

### PR DESCRIPTION
**Pull request by BethYates**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/genomenote/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by BethYates** [comment]
This is the merge of the dev branch into the public-dev branch. I've also added the genome metadata sub workflow into the main pipeline

**Review by priyanka-surana — CHANGES_REQUESTED** [review]
I am yet to test the pipeline. But so far it looks good. Some minor comments and a general suggestion – can we simplify this parameter name `genome_metadata_file_template`?

File: README.md [hunk 1]
```diff
@@ -1,99 +1,89 @@
...
 ## Credits
 
-sanger-tol/genomenote was originally written by @priyanka-surana.
+sanger-tol/genomenote was originally written by [Priyanka Surana](https://github.com/priyanka-surana).
 
-We thank the following people for their extensive assistance in the development of this pipeline:
+We thank the following people for their assistance in the development of this pipeline:
 
-– @mcshane and @yumisims for providing software/tool details
+- [Matthieu Muffato](https://github.com/muffato)
```
[the remark this answers is not in the document]
Comment (BethYates) [reply README.md:76]: Done

File: conf/test_full.config
[the remark this answers is not in the document]
Comment (BethYates) [reply conf/test_full.config:1]: Yes, changes added.

File: workflows/genomenote.nf [hunk 2]
```diff
@@ -9,10 +9,35 @@ def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
...
-else if (params.input && params.project) { inputs = [ params.input, params.project ] }
-else { exit 1, 'Input not specified. Please include either a samplesheet or Tree of Life organism and project IDs' }
+if (params.input)     { ch_input = Channel.fromPath(params.input) } else { exit 1, 'Input samplesheet not specified!' }
+if (params.fasta)     { ch_fasta = Channel.fromPath(params.fasta) } else { exit 1, 'Genome fasta not specified!' }
+if (params.binsize)   { ch_bin   = Channel.of(params.binsize)     } else { exit 1, 'Bin size for cooler/cload not specified!' }
+if (params.kmer_size) { ch_kmer  = Channel.of(params.kmer_size)   } else { exit 1, 'Kmer library size for fastk not specified' }
+
+if (params.assembly && params.taxon_id && params.bioproject && params.biosample) { metadata_inputs = [ params.assembly, params.taxon_id, params.bioproject, params.biosample ] }
+else { exit 1, 'Metadata input not specified. Please include an assembly accession, a taxon id, a bioproject accession and a biosample_accession' }
+
```
[the remark this answers is not in the document]
Comment (BethYates) [reply workflows/genomenote.nf]: Changes made

File: workflows/genomenote.nf [hunk 3]
```diff
@@ -51,53 +81,99 @@ workflow GENOMENOTE {
 
     ch_versions = Channel.empty()
 
+    //
+    // SUBWORKFLOW: Read in template of data files to fetch, parse these files and output a list of genome metadata params
+    
+    Channel.of ( metadata_inputs ).set { ch_metdata_input }
+    
+    ch_file_list = Channel.fromPath(params.genome_metadata_file_template)   
+
```
[the remark this answers is not in the document]
Comment (BethYates) [reply workflows/genomenote.nf]: Changes made

**Review by priyanka-surana — APPROVED** [review]
Looks good. We can work on CITATIONS, etc. closer to release. Thank you for making the changes.

**Pull request merged**